public/scripts/extensions/tts/index.js

```diff
@@ -638,6 +638,22 @@ async function processTtsQueue() {
             : text.replaceAll('*', '').trim(); // remove just the asterisks
     }
 
+    if (extension_settings.tts.apply_regex) {
+        const regexPattern = extension_settings.tts.regex_pattern || '/[^a-zA-Z0-9\\s.,!?;:\'"()-]+/g';
+        try {
+            const regex = regexFromString(regexPattern);
+            if (!regex) {
+                console.error('Invalid regex pattern:', regexPattern);
+            } else {
+                text = text.replace(regex, '').trim();
```

Comment:
You likely could combine two replace and save on one trim.
```suggestion
                text = text.replace(regex, '').replace(/\s+/g, ' ').trim();
```